src/internal/agent/state/state.go

```diff
@@ -0,0 +1,36 @@
+package state
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/defenseunicorns/zarf/src/internal/message"
+	"github.com/defenseunicorns/zarf/src/types"
+)
+
+const zarfStatePath = "/etc/zarf-state/state"
+
+// GetZarfStateFromAgentPod reads the state json file that was mounted into the agent pods.
+func GetZarfStateFromAgentPod() (types.ZarfState, error) {
+	var zarfState types.ZarfState
+
+	// Read the state file
+	stateFile, err := os.ReadFile(zarfStatePath)
+	if err != nil {
+		message.Warnf("Unable to read the zarfState file within the zarf-agent pod.")
```

Comment:
Nit; should we add the `zarfStatePath` to the error message (here and below)?

People might be able to exec into the pod and look to see if it's there without having to come back to the source code to figure out where the state-file should be.